wasm-runtime/polkadot/src/codec/slicable.rs

```diff
@@ -45,14 +47,14 @@ pub trait Slicable: Sized {
 pub trait NonTrivialSlicable: Slicable {}
 
 impl<T: EndianSensitive> Slicable for T {
-	fn set_as_slice<F: FnOnce(&mut [u8]) -> bool>(fill_slice: F) -> Option<Self> {
+	fn set_as_slice<F: Fn(&mut [u8], usize) -> bool>(fill_slice: &F) -> Option<Self> {
 		let size = mem::size_of::<T>();
 		let mut result: T = unsafe { mem::zeroed() };
 		let result_slice = unsafe {
 			let ptr = &mut result as *mut _ as *mut u8;
 			slice::from_raw_parts_mut(ptr, size)
```

Comment:
if size is zero then this is UB

Reply:
the size is never zero - `EndianSensitive` is only implemented for "real" types.  if there's a static check to ensure this is the case, great. Or some other way of preventing external code from using `EndianSensitive`. For now i'll make a runtime-assert.